WebSocket's binaryType setter should not throw

annevk opened the pull request:
#### 6e89039cfb2d2f7701a6c33cab9c24055d9291f6
<pre>
WebSocket&apos;s binaryType setter should not throw
<a href="https://bugs.webkit.org/show_bug.cgi?id=256580">https://bugs.webkit.org/show_bug.cgi?id=256580</a>
rdar://109192086

Reviewed by Antti Koivisto.

Align WebSocket.idl with the WebSockets Standard. In particular:

- binaryType is now an enum
- protocol and extensions are no longer nullable (I don&apos;t think they ever returned null)

* LayoutTests/TestExpectations:
* LayoutTests/http/tests/websocket/tests/hybi/binary-type-expected.txt:
* LayoutTests/http/tests/websocket/tests/hybi/binary-type.html:
* LayoutTests/imported/w3c/web-platform-tests/websockets/binaryType-wrong-value.any.worker_wss-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/websockets/binaryType-wrong-value.any_wss-expected.txt:
* Source/WebCore/Modules/websockets/WebSocket.cpp:
(WebCore::WebSocket::setBinaryType):
(WebCore::WebSocket::didReceiveBinaryData):
(WebCore::WebSocket::binaryType const): Deleted.
* Source/WebCore/Modules/websockets/WebSocket.h:
(WebCore::WebSocket::binaryType const):
* Source/WebCore/Modules/websockets/WebSocket.idl:
* Source/WebCore/xml/XMLHttpRequest.h:
(WebCore::XMLHttpRequest::responseType const): Deleted.
(WebCore::XMLHttpRequest::readyState const): Deleted.

Inline these into their declaration as well, matching WebSocket&apos;s binaryType(), per a review nit.

Canonical link: <a href="https://commits.webkit.org/263963@main">https://commits.webkit.org/263963@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a91633eed46e0138ff86f664f8ffbafd859e71fe

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6244 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6438 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6617 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7814 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6576 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/6245 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6658 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6388 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/9461 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6356 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/6367 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5652 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/7885 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3840 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/5631 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/13523 "Failed to checkout and rebase branch from PR 13742") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5705 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5712 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/7958 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6193 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5066 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5604 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/5560 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1478 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9753 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5972 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

anttijk commented on Source/WebCore/Modules/websockets/WebSocket.h:
```diff
@@ -142,4 +141,9 @@ class WebSocket final : public RefCounted<WebSocket>, public EventTarget, public
+inline auto WebSocket::binaryType() const -> BinaryType
+{
+    return m_binaryType;
+}
```
This could be either in cpp or inlined with the declaration (being trivial).

annevk replied:
Thanks for the review!

I copied this from XMLHttpRequest. I'll see about inlining both in the declaration and see how that goes.

annevk committed:
WebSocket's binaryType setter should not throw
https://bugs.webkit.org/show_bug.cgi?id=256580
rdar://109192086

Reviewed by Antti Koivisto.

Align WebSocket.idl with the WebSockets Standard. In particular:

- binaryType is now an enum
- protocol and extensions are no longer nullable (I don't think they ever returned null)

* LayoutTests/TestExpectations:
* LayoutTests/http/tests/websocket/tests/hybi/binary-type-expected.txt:
* LayoutTests/http/tests/websocket/tests/hybi/binary-type.html:
* LayoutTests/imported/w3c/web-platform-tests/websockets/binaryType-wrong-value.any.worker_wss-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/websockets/binaryType-wrong-value.any_wss-expected.txt:
* Source/WebCore/Modules/websockets/WebSocket.cpp:
(WebCore::WebSocket::setBinaryType):
(WebCore::WebSocket::didReceiveBinaryData):
(WebCore::WebSocket::binaryType const): Deleted.
* Source/WebCore/Modules/websockets/WebSocket.h:
(WebCore::WebSocket::binaryType const):
* Source/WebCore/Modules/websockets/WebSocket.idl:
* Source/WebCore/xml/XMLHttpRequest.h:
(WebCore::XMLHttpRequest::responseType const): Deleted.
(WebCore::XMLHttpRequest::readyState const): Deleted.

Inline these into their declaration as well, matching WebSocket's binaryType(), per a review nit.

Canonical link: https://commits.webkit.org/263963@main
